### PR TITLE
add list rule with service_group_id test

### PR DESCRIPTION
**Pull request by cam-d12**
## PR summary

This PR adds CBR support for the new service_group_id parameter in rules. 

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Current vs new behavior  

No current behavior should be modified. Adds the further control to be able to apply rules to service groups

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

**Review by soohoowoohoo — COMMENTED** [review]
Looks good.

**Review by soohoowoohoo — CHANGES_REQUESTED** [review]
A few comments.

File: contextbasedrestrictionsv1/context_based_restrictions_v1_integration_test.go [hunk 1]
```diff
@@ -646,6 +646,75 @@ var _ = Describe(`ContextBasedRestrictionsV1 Integration Tests`, func() {
...
+		It(`ListRules(listRulesOptions *ListRulesOptions) with service_group_id (201)`, func() {
+			listRulesOptions := &contextbasedrestrictionsv1.ListRulesOptions{
+				AccountID:     core.StringPtr(InvalidID),
+				TransactionID: getTransactionID(),
+			}
+
+			ruleList, response, err := contextBasedRestrictionsService.ListRules(listRulesOptions)
+
+			Expect(err).To(Not(BeNil()))
+			Expect(response.StatusCode).To(Equal(201))
```
Comment (soohoowoohoo) [inline contextbasedrestrictionsv1/context_based_restrictions_v1_integration_test.go:701]: Should be a 200

**Review by soohoowoohoo — COMMENTED** [review]
LGTM

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**